### PR TITLE
Add a custom pipeline

### PR DESCRIPTION
**Pull request by peuanthinsan**

### CODE REVIEW

File: src/containers/pipelines/AddModal/index.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      title="Add Pipelines"
+      visible
+      footer={null}
+      onOk={handleOk}
+      onCancel={handleCancel}
+    >
+      <StyledH5>Select a pipeline type</StyledH5>
+      <StyledPipelineGrid>
+        <StyledPipelineCard>
+          <Button onClick={null}>
```
Comment (divijb) [inline src/containers/pipelines/AddModal/index.js:41]:
```suggestion
          <Button>
```

Comment (divijb) [reply src/containers/pipelines/AddModal/index.js:41]: or if linter complains about needing an onClick use `() => {}` but not null

File: src/containers/pipelines/index.js [hunk 2]
```diff
@@ -7,33 +7,41 @@ import { getPipelines as getPipelinesAction } from 'actions/pipelines';
...
-          }
-        </StyledPipelines>
-      </>
-    );
-  }
+  }, []);
+
+  const [addModalVisible, setAddModalVisible] = useState(false);
+
+  const openModal = () => setAddModalVisible(true);
```
Comment (divijb) [inline src/containers/pipelines/index.js]: could we get away with `toggleModal` ?
Unless we need to take specific actions during open/close

**Review by divijb — COMMENTED** [review]
A few minor items. 
And also the build failed. It's probably the linter. Please be sure to run before commiting

**Comment by peuanthinsan** [comment]
thanks! it was a  warning on a missing dependency in useEffect

**Review by krohling — APPROVED** [review]

**Pull request merged**